src/H5public.h

```diff
@@ -368,6 +343,13 @@ typedef struct H5_alloc_stats_t {
  */
 typedef void (*H5_atclose_func_t)(void *ctx);
 
+/* API adapter header (defines H5_DLL, etc.) */
+#include "H5api_adpt.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
Not crucial for the gcc macros. Just moves the extern "C" and API adapter closer to where they are actually needed.